dagger/utilities/config_validator.py

```diff
@@ -98,10 +98,14 @@ def init_attributes_once(cls, orig_cls: str) -> None:
 
         cls.init_attributes(orig_cls)
         if parent_class.__name__ != "ConfigValidator":
-            cls.config_attributes[cls.__name__] = (
-                cls.config_attributes[parent_class.__name__]
-                + cls.config_attributes[cls.__name__]
-            )
+            parent_attributes = cls.config_attributes[parent_class.__name__]
+            current_attributes = cls.config_attributes[cls.__name__]
+
+            merged_attributes = {attr.name: attr for attr in parent_attributes}
+            for attr in current_attributes:
+                merged_attributes[attr.name] = attr
+
+            cls.config_attributes[cls.__name__] = list(merged_attributes.values())
```

Comment:
Nice! could we add the comment here to explain what it is doing?
it was hard for me to understand without reading the PR description.

Reply:
Done